### PR TITLE
MongooseIM 3.7.0

### PR DESCRIPTION
**Pull request by michalwski**
This PR is the final step for MongooseIM 3.7.0

Prosed changes:
* Update to CHANGELOG.md fie
* Update to VERSION file
* Update to README files

### CODE REVIEW

**Comment by mongoose-im** [comment]
[8261.1](https://travis-ci.org/esl/MongooseIM/jobs/687479346) / Erlang 22.0 / small_tests / bb183317cfbe608717595d8b10a21e5956e71ea8
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2744/8261/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2744/8261/small_tests.22.0/small/ct_run.mongooseim@localhost.2020-05-15_15.30.18/lib.mongooseim.logs/run.2020-05-15_15.30.18/suite.log.html)

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,74 @@
...
+
+## Highlights
+
+- Authentication methods improvements
+    - Support for `SCRAM-SHA-256`, `SCRAM-SHA-512`, `SCRAM-SHA-224`, `SCRAM-SHA-384` authentication methods
+    - Support for channel binding for all the SCRAM based authentication methods
+- Support for [XEP-0424: Message Retraction][XEP-0424]
+- Support for [Proxy Protocol]
+
+There were some changes to the database schema so please take a look at [Migration guide](doc/migrations/3.6.0_3.7.0.md) when upgrading from a previous version.
```
Comment (NelsonVides) [inline CHANGELOG.md:11]:
```suggestion
There were some changes to the database schema so please take a look at the [Migration guide](doc/migrations/3.6.0_3.7.0.md) when upgrading from a previous version.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,74 @@
...
+   - Stanza count (#2672)
+- Chat markers cache (#2676)
+- `mongooseimctl bootstrap` command which can be run to execute user defined init scripts (#2692)
+   - Support for templating the config files before MongooseIM starts (#2712)
+- `c2s_remote_hook_call` to run a hook in the context of the c2s process (#2700)
+- `mongooseimctl http_upload` command to help debug HTTP file upload configuration (#2708)
+
+### Changed
+
+- All hooks have been wrapped in `mongooseim_hooks` module with proper specs (#2642)
```
Comment (NelsonVides) [inline CHANGELOG.md:37]:
```suggestion
- All hooks have been wrapped in the `mongooseim_hooks` module with proper specs (#2642)
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,74 @@
...
+- Stream error reporting when the server receives unexpected stanza on given stream state (#2667)
```
Comment (NelsonVides) [inline CHANGELOG.md:38]:
```suggestion
- Stream error reporting when the server receives unexpected stanza on a given stream state (#2667)
```

**Comment by mongoose-im** [comment]
[8276.1](https://travis-ci.org/esl/MongooseIM/jobs/688715241) / Erlang 22.0 / small_tests / c35c8edaf1920a2a2a7230b4814ca8890c119310
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2744/8276/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2744/8276/small_tests.22.0/small/ct_run.mongooseim@localhost.2020-05-19_09.05.36/lib.mongooseim.logs/run.2020-05-19_09.05.37/suite.log.html)

**Comment by mongoose-im** [comment]
[8282.1](https://travis-ci.org/esl/MongooseIM/jobs/688733724) / Erlang 22.0 / small_tests / c1853a29ba23e93d0600bad94ec9095eb2a2b9fa
Reports [root](http://esl.github.io/mongooseim-ct-reports/s3_reports.html?prefix=PR/2744/8282/small_tests.22.0) / [small](https://mongooseim-ct-results.s3-eu-west-1.amazonaws.com/PR/2744/8282/small_tests.22.0/small/ct_run.mongooseim@localhost.2020-05-19_11.50.25/lib.mongooseim.logs/run.2020-05-19_11.50.25/suite.log.html)

**Pull request merged**